pandas/tests/test_format.py

```diff
@@ -2986,6 +2986,29 @@ def test_format(self):
         self.assertEqual(result[0], " 12")
         self.assertEqual(result[1], "  0")
 
+    def test_output_significant_digits(self):
+        # Issue #9764
+        
+        # In case default display precision changes:
+        saved_option=pd.get_option('display.precision')
+        pd.set_option('display.precision', 7)
```

Comment:
Can you set this with the context manager `pd.option_context()` instead? That's safer than setting a global variable.